ui/robolectric/BasicSample/app/src/test/java/com/example/android/testing/robolectric/BasicSample/ShowTextActivityTest.java

```diff
@@ -0,0 +1,38 @@
+package com.example.android.testing.robolectric.BasicSample;
+
+import android.content.Intent;
+import android.widget.TextView;
+
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+/**
+ * Robolectric tests for {@link ShowTextActivity}.
+ */
+@RunWith(RobolectricTestRunner.class)
+public class ShowTextActivityTest {
+
+    private static final String MESSAGE = "Example";
+
+    private ShowTextActivity activity;
+
+    @Before
+    public void setUp() {
+        Intent intent = new Intent(RuntimeEnvironment.application, ShowTextActivity.class);
+        intent.putExtra(ShowTextActivity.KEY_EXTRA_MESSAGE, MESSAGE);
+        activity = Robolectric.buildActivity(ShowTextActivity.class, intent).setup().get();
```

Comment:
Robolectric.setupActivity()

Reply:
I don't think it works with an intent does it?